client/extensions/woocommerce/state/sites/products/reducer.js

```diff
@@ -139,3 +143,9 @@ function setLoading( state, page, newStatus ) {
 	isLoading[ page ] = newStatus;
 	return isLoading;
 }
+
+function setError( state, page, newStatus ) {
+	const isError = state.isError && { ...state.isError } || {};
```

Comment:
I think `{ ...undefined }` should return {}, right? So I think you can remove the && and || here.